packages/ts/react-grid/src/autocrud.tsx

```diff
@@ -1,58 +1,61 @@
 import { AutoGrid } from './autogrid';
 import type { CrudService } from './crud';
+import { getProperties } from './property-info';
 
 export type AutoCrudProps<TItem> = Readonly<{
   service: CrudService<TItem>;
   model: DetachedModelConstructor<AbstractModel<TItem>>;
 }>;
 
-interface AutoCrudContextType<TItem> {
-  item: TItem | undefined;
-}
-
-export const AutoCrudContext = createContext<AutoCrudContextType<any>>({ item: undefined });
-
 export function ExperimentalAutoCrud<TItem>({ service, model }: AutoCrudProps<TItem>): JSX.Element {
   const [item, setItem] = useState<TItem | undefined>(undefined);
   const [refreshTrigger, setRefreshTrigger] = useState(0);
 
   return (
     <>
-      <HorizontalLayout>
-        <AutoGrid
-          refreshTrigger={refreshTrigger}
-          service={service}
-          model={model}
-          onActiveItemChanged={(e) => {
-            const activeItem = e.detail.value;
-            (e.target as GridElement).selectedItems = activeItem ? [activeItem] : [];
-          }}
-          onSelectedItemsChanged={(e) => {
-            if (e.detail.value.length === 0) {
-              setItem(undefined);
-            } else {
-              const selectedItem = e.detail.value[0];
-              setItem({ ...selectedItem });
-            }
-          }}
-        ></AutoGrid>
-        <ExperimentalAutoForm
-          disabled={!item}
-          service={service}
-          model={model}
-          item={item}
-          afterSubmit={({ item: submittedItem }) => {
-            setItem(submittedItem);
-            // Trigger grid data refresh
-            setRefreshTrigger(refreshTrigger + 1);
-          }}
-        ></ExperimentalAutoForm>
-      </HorizontalLayout>
+      <AutoCrudContext.Provider
+        value={{ service, properties: getProperties(model), refreshGrid: () => setRefreshTrigger(refreshTrigger + 1) }}
+      >
+        <HorizontalLayout>
+          <AutoGrid
+            refreshTrigger={refreshTrigger}
+            service={service}
+            model={model}
+            onActiveItemChanged={(e) => {
+              const activeItem = e.detail.value;
+              (e.target as GridElement).selectedItems = activeItem ? [activeItem] : [];
+            }}
+            onSelectedItemsChanged={(e) => {
+              if (e.detail.value.length === 0) {
+                setItem(undefined);
+              } else {
+                const selectedItem = e.detail.value[0];
+                setItem({ ...selectedItem });
+              }
+            }}
+            customColumns={[<GridColumn autoWidth renderer={DeleteButton}></GridColumn>]}
+          ></AutoGrid>
+          <ExperimentalAutoForm
+            disabled={!item}
+            service={service}
+            model={model}
+            item={item}
+            afterSubmit={({ item: submittedItem }) => {
+              setItem(submittedItem);
+              // Trigger grid data refresh
+              setRefreshTrigger(refreshTrigger + 1);
```

Comment:
Yes but it cannot right now as both the context definition and user is in the same file

Reply:
So is it for the outside world to call?

Reply:
So is it for the outside world to call?

Reply:
Well, TBH, I don't understand the point for defining that `refreshGrid` then, since it is not callable inside neither from outside, but I guess you had something in mind when adding it, possibly for the next PRs. 
We can call someone else with more knowledge in React and TS, or you can resolve this discussion.